File: backend/src/main/java/io/tingkai/money/entity/Fund.java
package io.tingkai.money.entity;

import java.time.LocalDateTime;
import java.util.UUID;

import io.tingkai.money.constant.DatabaseConstants;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = DatabaseConstants.TABLE_FUND)
public class Fund {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	protected UUID id;
	@Column(unique = true)
	protected String code;
	@Column(unique = true)
	protected String symbol;
	protected String name;
	@Column(unique = true)
	protected String isinCode;
	protected LocalDateTime offeringDate;
	protected String currency;
	protected String description;
}

File: python/src/entity/Fund.py
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.schema import UniqueConstraint
import uuid

from entity.BaseEntity import BaseEntity, db

class Fund(BaseEntity):
    __tablename__ = 'fund'

    id = db.Column(UUID(as_uuid = True), primary_key = True, default = uuid.uuid4)
    code = db.Column(db.String, nullable = False)
    symbol = db.Column(db.String)
    name = db.Column(db.String, nullable = False)
    isin_code = db.Column(db.String, nullable = False)
    offering_date = db.Column(db.DateTime, nullable = False)
    currency = db.Column(db.String, nullable = False)
    description = db.Column(db.String)

    UniqueConstraint('code', name = 'fund_code_key')
    UniqueConstraint('isin_code', name = 'fund_isin_code_key')
